provider/activitypub/mastodon/client.go

```diff
@@ -32,29 +33,50 @@ func (c *client) GetKafkaConsumer() *kgo.Client {
 	return c.kafkaConsumer
 }
 
-func NewClient(endpoint string, kafkaTopic string) (Client, error) {
-	var (
-		instance = client{
-			httpClient: &http.Client{
-				Timeout: DefaultTimeout,
-			},
-			encoder:  form.NewEncoder(),
-			attempts: DefaultAttempts,
-		}
-	)
+// extractKafkaBroker extracts the host:port part from the given endpoint
+func extractKafkaBroker(endpoint string) string {
+	kafkaBroker := endpoint
 
-	// extract the host:port part for setting kafka consumer service
-	trimmedHostPort := strings.TrimPrefix(endpoint, "http://")
+	// Remove the protocol (e.g., "http://") if present
+	if strings.Contains(kafkaBroker, "://") {
```

Comment:
Maybe using `url` pkg would be better.

Reply:
Since we cannot determine how the user connects to Kafka, we directly use the endpoint as the broker here